uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlement-change-log.component.js

```diff
@@ -0,0 +1,319 @@
+        });
+    }
+
+    /**
+     * Selects the entitlement rows to be highlighlighted based on the one
+     * that has comments.
+     */
+    function highlightEntitlementWithComments () {
+      for (var i = vm.changeLogRows.length - 1; i >= 0; i--) {
+        var changeLogRow = vm.changeLogRows[i];
+        var entitlementComments = _.pluck(changeLogRow.entitlements, 'comment');
+        var validEntitlementComments = _.compact(entitlementComments).length;
+
+        if (validEntitlementComments === 1) {
+          var commentIndex = _.findIndex(entitlementComments, 'length');
+
+          changeLogRow.highlightedEntitlement = changeLogRow
+            .entitlements[commentIndex];
+        } else if (validEntitlementComments > 1) {
+          splitEntitlementCommentsIntoMultipleRows(i);
+        }
+      }
+    }
+
+    /**
+     * Returns the current entitlements for the contact and period and maps
+     * them into entitlement logs format.
+     *
+     * @return {Promise}
+     */
+    function getCurrentEntitlementsLog () {
+      return Entitlement.all({
+        contact_id: vm.contactId,
+        period_id: vm.periodId
+      }, false)
+      .then(function (currentEntitlements) {
+        return currentEntitlements.map(function (entitlement) {
+          return {
+            'comment': entitlement.comment,
+            'contact_id': entitlement.contact_id,
+            'created_date': entitlement.created_date,
+            'editor_id': entitlement.editor_id,
+            'entitlement_amount': entitlement.value,
+            'entitlement_id': entitlement.id,
+            'entitlement_id.type_id': entitlement.type_id
+          };
+        });
+      });
+    }
+
+    /**
+     * Each change log row consist of their creation date and a list of
+     * entitlements that are sorted in the same order as the absence types that
+     * are displayed in the table's header. The calculation units are also added
+     * to the entitlements in order to display if the change was in hours or days.
+     *
+     * @param {Array} entitlements - List of entitlement changes for a specific date.
+     * @param {String} createdDate - The date of creation for the entitlements
+     * provided.
+     * @return {Object}
+     */
+    function getChangeLogRow (entitlements, createdDate) {
+      var indexedEntitlements = _.indexBy(entitlements, 'entitlement_id.type_id');
+      var sortedEntitlements = vm.absenceTypes.map(function (absenceType) {
+        var entitlement = indexedEntitlements[absenceType.id];
+
+        return _.extend({
+          calculation_unit: absenceType['calculation_unit.name']
+        }, entitlement);
+      });
+
+      return {
+        date: moment(createdDate),
+        entitlements: sortedEntitlements
+      };
+    }
+
+    /**
+     * Helper function that groups entitlements by their creation date. The
+     * creation date is rounded to the nearest 15 second interval (0, 15, 30,
```

Comment:
Was the number `15` confirmed with anyone?

Reply:
Rounding will be removed.